src/vcpkg/commands.portsdiff.cpp

```diff
@@ -25,78 +26,112 @@ namespace
-
-        RedirectedProcessLaunchSettings settings;
-        settings.environment = get_clean_environment();
-        flatten(cmd_execute_and_capture_output(paths.git_cmd_builder(dot_git_dir, temp_checkout_path)
-                                                   .string_arg("checkout")
-                                                   .string_arg(git_commit_id)
-                                                   .string_arg("-f")
-                                                   .string_arg("-q")
-                                                   .string_arg("--")
-                                                   .string_arg(checkout_this_dir)
-                                                   .string_arg(".vcpkg-root"),
-                                               settings),
-                Tools::GIT)
-            .value_or_exit(VCPKG_LINE_INFO);
-        flatten(cmd_execute_and_capture_output(
-                    paths.git_cmd_builder(dot_git_dir, temp_checkout_path).string_arg("reset"), settings),
-                Tools::GIT)
-            .value_or_exit(VCPKG_LINE_INFO);
-
-        OverlayPortIndexEntry ports_at_commit_index(OverlayPortKind::Directory, temp_checkout_path / ports_dir_name);
+        const auto& builtin_ports_directory = paths.builtin_ports_directory();
+        auto maybe_builtin_ports_prefix = git_prefix(context, git_exe, builtin_ports_directory);
+        const auto builtin_ports_prefix = maybe_builtin_ports_prefix.get();
+        if (!builtin_ports_prefix)
+        {
+            return nullopt;
+        }
+
+        // remove the trailing slash
+        if (!builtin_ports_prefix->empty())
+        {
+            builtin_ports_prefix->pop_back();
+        }
+
+        const auto temp_checkout_path = paths.buildtrees() / temp_name;
+        if (!git_read_tree(context,
+                           fs,
+                           git_exe,
+                           GitRepoLocator{GitRepoLocatorKind::CurrentDirectory, builtin_ports_directory},
+                           temp_checkout_path,
+                           fmt::format("{}:{}", git_commit_id, *builtin_ports_prefix)))
+        {
+            return nullopt;
+        }
+
+        OverlayPortIndexEntry ports_at_commit_index(OverlayPortKind::Directory, temp_checkout_path);
         std::map<std::string, const SourceControlFileAndLocation*> ports_at_commit;
-        ports_at_commit_index.try_load_all_ports(fs, ports_at_commit).value_or_exit(VCPKG_LINE_INFO);
-        fs.remove_all(temp_checkout_path, VCPKG_LINE_INFO);
+        auto maybe_loaded_all_ports = ports_at_commit_index.try_load_all_ports(fs, ports_at_commit);
+        if (!maybe_loaded_all_ports)
+        {
+            context.report(DiagnosticLine{DiagKind::None, std::move(maybe_loaded_all_ports).error()});
+            return nullopt;
+        }
+
+        if (!fs.remove_all(context, temp_checkout_path))
```

Comment:
Should we do this cleanup in the "failed to load all ports" case?

Reply:
Upon further reflection in that case I'd like to leave it behind for user diagnostics